ironforgedbot/commands/admin/cmd_sync_members.py

```diff
@@ -40,7 +54,7 @@ def sync_members(guild: discord.Guild, storage: IngotsStorage) -> list[str]:
 
     # Then, get all current entries from storage.
     try:
-        existing = storage.read_members()
+        existing = STORAGE.read_members()
```

Comment:
tbh I'm very against global variables: pain to ensure state, mock, etc.

Reply:
So these are not global variables. This is the singleton pattern in Python. Not very obvious I know, but what this does is essentially exactly the same as we had before, except now we are not passing around references to instances everywhere.

As for testing, I've found is pretty straight forward to mock etc, take a look at some of the tests to get an idea of how this works

Reply:
It is not a singleton, unless you define entire package as one. pep has example implementation of singleton https://peps.python.org/pep-0318/#examples, alternatively borg can be used. This is a module-shared variable. Using this prevents you from forking, if we ever need one.

Reply:
You're right, this is not a traditional singleton.

I think they call it a 'module level singleton'. Functionally it does the exact same thing as a singleton class would for our current use-case.I did it this way as I wanted a nice clean way to fail out and exit during bot startup if there was any issue with connecting, rather than waiting for the first time it is used.

Reply:
You're right, this is not a traditional singleton.

I think they call it a 'module level singleton'. Functionally it does the exact same thing as a singleton class would for our current use-case.

I did it this way as I wanted a nice clean way to fail out early and exit during bot startup if there was any issue with connecting, rather than waiting for the first time it is used to throw an error.

I think I will address this in a separate PR and try and use a more structured class based approach.